src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  alive() {
+    const { name, health, hidden } = this;
+
+    if (health > 0 && hidden !== undefined) {
+      return `name: ${name}, health: ${health}, hidden: ${hidden}`;
+    } else if (health > 0 && hidden === undefined) {
+      return `name: ${name}, health: ${health}`;
+    }
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast instanceof Herbivore && !beast.hidden) {
+      beast.health -= 50;
+      Carnivore.removeDeadAnimals();
```

Comment:
There's no need to launch this method after every bite. It's only needed when an animal's health is not greater than 0.